save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/utils/SetupUtils.kt

```diff
@@ -4,21 +4,80 @@
 
 package com.saveourtool.save.demo.agent.utils
 
+import com.saveourtool.save.core.logging.logDebug
+import com.saveourtool.save.core.logging.logError
+import com.saveourtool.save.core.logging.logInfo
+import com.saveourtool.save.core.utils.ExecutionResult
+import com.saveourtool.save.core.utils.ProcessBuilder
 import com.saveourtool.save.demo.agent.DemoConfiguration
+import com.saveourtool.save.utils.*
+
+import io.ktor.server.application.*
+import okio.Path.Companion.toPath
+
+import kotlinx.coroutines.*
+
+private const val SETUP_SH_TIMEOUT_MILLIS = 5000L
+private const val SETUP_SH_LOGS_FILENAME = "setup.logs"
+private const val CWD = "."
 
 /**
  * Download all the required files from save-demo
  *
  * @param demoUrl url to save-demo
  * @param demoConfiguration all the information required for tool download
+ * @throws IllegalStateException when it was caught from [downloadDemoFiles]
  */
-fun setupEnvironment(demoUrl: String, demoConfiguration: DemoConfiguration) {
-    downloadDemoFiles(demoUrl, demoConfiguration)
-    executeSetupSh()
+@OptIn(ExperimentalStdlibApi::class)
+suspend fun setupEnvironment(demoUrl: String, demoConfiguration: DemoConfiguration) {
+    logInfo("Setting up the environment...")
+
+    try {
+        downloadDemoFiles(demoUrl, demoConfiguration)
+    } catch (e: IllegalStateException) {
+        logError("Error while downloading files to agent.")
```

Comment:
Does it make sense to also `log e.message ?: e.toString()`?

Reply:
Ok, sure

Reply:
Done